src/download.rs

```diff
@@ -794,11 +948,87 @@ impl<'a> AssetDownloader<'a> {
                 return Err(anyhow!("Could not download asset: {err}"));
             }
         }
+
+        if self.ctx.args.one_html_file {
+            // We now need to write back the data URLs into the JSON.
+            let mut case_map: HashMap<u32, &mut Case> =
+                cases.iter_mut().map(|x| (x.id(), x)).collect();
+            for asset in downloads.iter().filter(|x| x.1.is_ok()).map(|x| x.0) {
+                Self::rewrite_data(asset, &mut case_map, site_data)?;
+            }
+        }
         Ok(())
     }
 
-    /// Returns the default sprites that are actually used in the case,
-    /// based on the given [profiles] and [`used_sprites`].
+    pub(crate) fn rewrite_data(
+        data_asset: &AssetDownload,
+        case_map: &mut HashMap<u32, &mut Case>,
+        site_data: &mut SiteData,
+    ) -> Result<()> {
+        let data_url = data_asset
+            .data_url
+            .clone()
+            .expect("Data URL must be present");
+        match &data_asset.json_ref.source {
+            JsonSource::CaseData(id) => {
+                let data = &mut case_map
+                    .get_mut(id)
+                    .expect("case with ID must be present")
+                    .case_data;
+                // Modify pointee to value with data URL.
+                *data
+                    .pointer_mut(&data_asset.json_ref.pointer)
+                    .expect("pointer must be valid") = Value::String(data_url);
+            }
+            JsonSource::DefaultPlaces => {
+                // We skip the root part of the string (i.e., "/{id}/rest...") that would be empty.
+                let mut pointer_parts = data_asset.json_ref.pointer.splitn(3, '/').skip(1);
+                let id = pointer_parts.next().unwrap();
+                let pointer = format!("/{}", pointer_parts.next().unwrap());
+                *site_data
+                    .default_data
+                    .default_places
+                    .get_mut(&id.parse()?)
+                    .expect("default place ID must be valid")
+                    .pointer_mut(&pointer)
+                    .expect("pointer must be valid") = Value::String(data_url);
```

Comment:
The method returns a `Result`, but you use `?` only sometimes and call `expect()` other times. Is this intentional?

Reply:
You're right, as this consists purely of programming errors (see other comment above), I've changed the `?` here to `expect` and changed the return type accordingly. If the ID cannot be parsed, that's the fault of my script extracting it incorrectly, so it should panic.